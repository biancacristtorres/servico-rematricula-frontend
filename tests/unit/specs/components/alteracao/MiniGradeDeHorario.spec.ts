import _ from 'lodash';
import {
  MiniGradeDeHorarioFactory,
} from '../../../factory/components/alteracao/MiniGradeDeHorarioFactory';

describe('Componente de MiniGradeDeHorario', () => {
  const factory = new MiniGradeDeHorarioFactory();

  it('snapshot', async (done) => {
    factory.dadoQueExistamHorarios();

    await factory.montarComponente();

    expect(factory.componente)
      .toMatchSnapshot();

    done();
  });

  it('A mini grade está em estado de loading', async (done) => {
    factory.dadoQueEstaAcontecendoUmaAlteracaoDeTurma();

    await factory.montarComponente();
    const loading = factory.componente.find('#loading-mini-grade');

    expect(loading.exists()).toBeTruthy();
    done();
  });


  it('A mini grade não está em estado de loading', async (done) => {
    factory.dadoQueNaoEstaAcontecendoUmaAlteracaoDeTurma();

    await factory.montarComponente();
    const loading = factory.componente.find('#loading-mini-grade');

    expect(loading.exists()).toBeFalsy();
    done();
  });

});
