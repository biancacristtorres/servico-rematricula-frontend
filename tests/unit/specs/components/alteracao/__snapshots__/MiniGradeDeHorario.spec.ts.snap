// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente de MiniGradeDeHorario snapshot 1`] = `
<div>
  <div class="container-tabela">
    <vcontainer-stub tag="div" class="container-grade box pa-0 caption">
      <vlayout-stub tag="div" row="">
        <vflex-stub tag="div" xs2="" class="display-table celula">
          <div class="display-cell">

          </div>
        </vflex-stub>
        <vflex-stub tag="div" xs2="" class="display-table celula">
          <div class="display-cell">
            S
          </div>
        </vflex-stub>
        <vflex-stub tag="div" xs2="" class="display-table celula">
          <div class="display-cell">
            T
          </div>
        </vflex-stub>
        <vflex-stub tag="div" xs2="" class="display-table celula">
          <div class="display-cell">
            Q
          </div>
        </vflex-stub>
        <vflex-stub tag="div" xs2="" class="display-table celula">
          <div class="display-cell">
            Q
          </div>
        </vflex-stub>
        <vflex-stub tag="div" xs2="" class="display-table celula">
          <div class="display-cell">
            S
          </div>
        </vflex-stub>
        <vflex-stub tag="div" xs2="" class="display-table celula">
          <div class="display-cell">
            S
          </div>
        </vflex-stub>
      </vlayout-stub>
      <vlayout-stub tag="div" row="" align-space-around="" justify-center="" fill-height="" class="align-content-center">
        <vflex-stub tag="div" text-xs-center="" xs2="" class="celula">
          <vlayout-stub tag="div" align-center="" justify-center="" row="" fill-height="">
            <div>
              20:55h
            </div>
          </vlayout-stub>
        </vflex-stub>
        <vflex-stub tag="div" xs2="" class="celula">
          <vlayout-stub tag="div" row="" wrap="" align-center="" justify-center="" fill-height="" class="pa-0">
            <vflex-stub tag="div" xs12="" align-center="" class="text-xs-center pa-1">
              <div style="width: 10px; height: 10px; background: rgb(12, 189, 133); border-radius: 50%; margin: 0px auto;"></div>
            </vflex-stub>
          </vlayout-stub>
        </vflex-stub>
        <vflex-stub tag="div" xs2="" class="celula">
          <vlayout-stub tag="div" row="" wrap="" align-center="" justify-center="" fill-height="" class="pa-0">
            <vflex-stub tag="div" xs12="" align-center="" class="text-xs-center pa-1">
              <div style="width: 10px; height: 10px; background: rgb(12, 189, 133); border-radius: 50%; margin: 0px auto;"></div>
            </vflex-stub>
          </vlayout-stub>
        </vflex-stub>
      </vlayout-stub>
    </vcontainer-stub>
  </div>
</div>
`;
